dateparser/search/search.py

```diff
@@ -171,27 +171,37 @@ def __init__(self):
         self.available_language_map = self.loader.get_locale_map()
         self.search = _ExactLanguageSearch(self.loader)
 
-    def detect_language(self, text, languages):
-        if isinstance(languages, (list, tuple, Set)):
+    @apply_settings
+    def detect_language(self, text, languages, settings=None, detect_languages_func=None):
+        if detect_languages_func:
+            detected_languages = detect_languages_func(text, settings=settings)
+            detected_languages = detected_languages or None
+            if not detected_languages:
+                if not settings.LANGUAGE_DETECTION_STRICT_USE:
+                    detected_languages = settings.DEFAULT_LANGUAGES or None
+                return detected_languages
+            return detected_languages[0]
 
+        if isinstance(languages, (list, tuple, Set)):
             if all([language in self.available_language_map for language in languages]):
                 languages = [self.available_language_map[language] for language in languages]
             else:
                 unsupported_languages = set(languages) - set(self.available_language_map.keys())
-                raise ValueError(
-                    "Unknown language(s): %s" % ', '.join(map(repr, unsupported_languages)))
+                raise ValueError("Unknown language(s): %s" % ', '.join(map(repr, unsupported_languages)))
         elif languages is not None:
             raise TypeError("languages argument must be a list (%r given)" % type(languages))
-
```

Comment:
I think is better to restore this empty line to separate better one `if-else` and the other `if-else`.